src/lib/markbind/src/parser.js

```diff
@@ -588,54 +576,53 @@ class Parser {
     });
   }
 
-  _rebaseReference(node, foundBase) {
+  /**
+   * Pre-renders the baseUrl of the provided node and its children according to
+   * the site they belong to, such that the final call to render baseUrl correctly
+   * resolves baseUrl according to the said site.
+   */
+  _rebaseReference(node) {
```

Comment:
Two changes here essentially:
- replace nested ifs with guard clauses
- removal of `foundBase` ( see the explanation in the OP )